src/rulesets/reader.ts

```diff
@@ -65,15 +65,17 @@ const createRulesetProcessor = (
         dereferenceInline: false,
         uriCache,
         async parseResolveResult(opts) {
-          try {
-            opts.result = parse(opts.result);
-          } catch {
-            // happens
-          }
-          return opts;
+          return new Promise<IUriParserResult>((resolve, reject) => {
```

Comment:
Is there any particular reason we need this promise here?
We have `async` fn in place already.

Reply:
Duh. :-/
...
Fixed. ;-)